sys/parsoid.js

```diff
@@ -380,6 +380,22 @@ PSP.getRevisionInfo = function(restbase, req) {
     });
 };
 
+/**
+ * Internal check to see if it's okay to re-render a particular title in
+ * response to a no-cache request.
+ * @param {Request}
+ * @return {boolean} Whether re-rendering this title is okay.
+ */
+PSP._okayToRerender = function(req) {
+    var blackList = this.options.rerenderBlacklist;
+    if (blackList) {
+        return !blackList[req.params.domain]
+            || !blackList[req.params.domain][req.params.title];
```

Comment:
Right, but malicious clients could get passed this check because of this assumption.